src/lib/registration/ui/addon_selection_base_dialog.rb

```diff
@@ -93,86 +101,84 @@ def details_widget
-          addons.size < 5
+      IMAGE_DIR = "/usr/share/YaST2/theme/current/wizard".freeze
+      IMAGES = {
+        "normal:on:enabled"   => "checkbox-on.png",
+        "normal:off:enabled"  => "checkbox-off.png",
+        # theme has no special images for disabled checkboxes
+        "normal:on:disabled"  => "checkbox-on.png",
+        "normal:off:disabled" => "checkbox-off.png",
+        "inst:on:enabled"     => "inst_checkbox-on.png",
+        "inst:off:enabled"    => "inst_checkbox-off.png",
+        "inst:on:disabled"    => "inst_checkbox-on-disabled.png",
+        "inst:off:disabled"   => "inst_checkbox-off-disabled.png"
+      }.freeze
+
+      # Make a simulation of a CheckBox displayed in a RichText
+      # @param id [String]
+      # @param label [String]
+      # @param selected [Boolean]
+      # @param enabled [Boolean]
+      # @param indented [Boolean]
+      # @return [String] a Value for a RichText
+      def richtext_checkbox(id:, label:, selected:, enabled:, indented: false)
+        selected = false unless enabled
+        if Yast::UI.TextMode
+          indent = "&nbsp;" * (indented ? 5 : 1)
+          check = selected ? "[x]" : "[ ]"
+          widget = "#{check} #{label}"
+          enabled_widget = enabled ? "<a href=\"#{id}\">#{widget}</a>" : widget
+          "#{indent}#{enabled_widget}<br>"
         else
-          true
-        end
+          indent = "&nbsp;" * (indented ? 7 : 1)
 
-        addons.each do |addon|
-          box.params.concat(addon_checkbox(addon, add_extra_spacing))
-        end
+          installation = !Yast::Mode.normal
+          image = (installation ? "inst:" : "normal:") +
+            (selected ? "on:" : "off:") + (enabled ? "enabled" : "disabled")
+          color = installation ? "white" : "black"
 
-        box
-      end
-
-      # create spacing around the addon checkbox so the layout looks better
-      # @param addon [Registration::Addon]
-      # @param extra_spacing [Boolean] add extra spacing (indicates enough space in UI)
-      # @return [Array<Yast::Term>] Return array with one or two elements for VBox
-      def addon_checkbox(addon, extra_spacing)
-        checkbox = Left(addon_checkbox_element(addon))
-
-        # usability help. If addon depends on something, then we get it
-        # immediatelly after parent, so indent it slightly, so it is easier visible
-        checkbox = HBox(HSpacing(2.5), checkbox) if addon.depends_on
-
-        res = [checkbox]
-        # add extra spacing when there are just few addons, in GUI always
-        res << VSpacing(0.7) if extra_spacing
-
-        res
+          check = "<img src='#{IMAGE_DIR}/#{IMAGES[image]}'></img>"
+          widget = "#{check} #{label}"
+          enabled_widget = if enabled
+            "<a href='#{id}' style='text-decoration:none; color:#{color}'>#{widget}</a>"
+          else
+            "<span style='color:grey'>#{widget}</span>"
+          end
+          "<p>#{indent}#{enabled_widget}</p>"
+        end
```

Comment:
my sandi metz part of brain see that it can be nice example for objects, like factory for checkbox representation, deciding if use text one or graphical one, with its elements and so on, but it is out of scope this pr.